File: src/main/java/net/cjsah/console/command/argument/BooleanArgument.java
package net.cjsah.console.command.argument;

import net.cjsah.console.command.StringReader;
import net.cjsah.console.command.context.CommandContext;
import net.cjsah.console.exceptions.CommandException;

public class BooleanArgument implements Argument<Boolean> {

    private BooleanArgument() {
    }

    public static BooleanArgument bool() {
        return new BooleanArgument();
    }

    public static Boolean getBoolean(CommandContext context, String name) {
        return context.getArgument(name, boolean.class);
    }

    @Override
    public Boolean parse(StringReader reader) throws CommandException {
        return reader.readBoolean();
    }
}

File: src/main/kotlin/net/cjsah/console/command/argument/BooleanArgument.kt
@file:Suppress("unused")

package net.cjsah.console.command.argument

import net.cjsah.console.command.StringReader
import net.cjsah.console.command.context.CommandContext

class BooleanArgument private constructor() : Argument<Boolean> {

    companion object {
        @JvmStatic
        fun bool() = BooleanArgument()

        @JvmStatic
        fun getBoolean(context: CommandContext, name: String) =  context.getArgument(name, Boolean::class.javaPrimitiveType!!)
    }

    override fun parse(reader: StringReader) = reader.readBoolean()
}